api/users/permissions.py

```diff
@@ -47,3 +49,63 @@ def has_permission(self, request, view):
     def has_object_permission(self, request, view, obj):
         assert isinstance(obj, OSFUser), f'obj must be a User, got {obj}'
         return not obj.is_registered
+
+
+class UserMessagePermissions(permissions.BasePermission):
+    """
+    Custom permission to allow only institutional admins to create certain types of UserMessages.
+    """
+    def has_permission(self, request, view) -> bool:
+        """
+        Validate if the user has permission to perform the requested action.
+        Args:
+            request: The HTTP request.
+            view: The view handling the request.
+        Returns:
+            bool: True if the user has the required permission, False otherwise.
+        """
+        if request.method != 'POST':
+            return False
+
+        user = request.user
+        if not user or user.is_anonymous:
+            return False
+
+        message_type = request.data.get('message_type')
+        if message_type == MessageTypes.INSTITUTIONAL_REQUEST:
+            return self._validate_institutional_request(request, user)
+
+        return False
+
+    def _validate_institutional_request(self, request, user: OSFUser) -> bool:
+        """
+        Validate the user's permissions for creating an `INSTITUTIONAL_REQUEST` message.
+        Args:
+            request: The HTTP request containing the institution ID.
+            user: The user making the request.
+        Returns:
+            bool: True if the user has the required permission.
+        """
+        institution_id = request.data.get('institution')
+        if not institution_id:
+            raise exceptions.ValidationError({'institution': 'Institution ID is required.'})
+
+        institution = self._get_institution(institution_id)
+
+        if not user.is_institutional_admin(institution):
+            raise exceptions.NotAuthenticated('You are not an admin of the specified institution.')
```

Comment:
This wouldn't be an appropriate use of a 401, which I presume this raises. It should be a 403 instead, because they don't have permission even though they are authenticated. So you should just be able to return false from this.